packages/@aws-cdk/cdk/lib/stack.ts

```diff
@@ -303,6 +335,43 @@ export class Stack extends Construct implements ITaggable {
     return Arn.format(components, this);
   }
 
+  /**
+   * Returnst the list of AZs that are availability in the AWS environment
+   * (account/region) associated with this stack.
+   *
+   * If the stack is environment-agnostic (either account and/or region are
+   * tokens), this property will return an array with 2 tokens that will resolve
+   * at deploy-time to the first two availability zones returned from CloudFormation's
+   * `Fn::GetAZs` intrinsic function.
+   *
+   * If they are not available in the context, returns a set of dummy values and
+   * reports them as missing, and let the CLI resolve them by calling EC2
+   * `DescribeAvailabilityZones` on the target environment.
+   */
+  public get availabilityZones() {
```

Comment:
I don't like this being here.

I don't feel like AvailabilityZones are so intrinsically linked to a stack that they need a first-class getter on `Stack`.

Yes, they need an environment (and so a stack) to anchor to, but all contexts do.

* What about `stack.ssmParameter(...)`, `stack.ami(...)`? Do those make sense? About as much as `stack.availabilityZones`.
* What about my fully serverless stack with APIGW and Lambda? Why is that stack saddled with a notion of `stack.availabilityZones`?

I liked it better where it was.

Reply:
As discussed in person, `stack.availabilityZones` are basically the CDK's version of the `Fn::GetAZs` intrinsic, and having this available from the `Stack` is in line with other CloudFormation intrinsics that are exposed from the Stack like `urlSuffix`, `partition`, etc.

@rix0rrr suggested to maybe make that available under `ec2.AvailabilityZones.fromLookup` or something like that, but we decided that for now we will leave it under `stack`.